UXPin/uxpin-merge-tools
# Simplify `@uxpin***` JSDoc annotation serialization

## Goal

Simplify the serialization of information passed by users as JSdoc annotations, using the `@uxpin***` keywords.
Currently the implementation is split between a lot of files that often contain only a very short function.

The goal is to avoid having to create new files when adding a new `@uxpin***` annotation.

The parsing logic is not the same for JS and TS components.
I tried to be consistent and create files with the same name, for both TS and JS.

So we have a folder `comments` inside both `typescript` and `javascript` folder, with 2 files:

- `jsdoc-uxpin-annotations.ts`: contains the function that extract value for a given tag
- `jsdoc-helpers.ts`: contains some helpers not specific to any tag


### Updated files for TS components

#### Before

```
steps/serialization/component/implementation/typescript/comments
├── jsdoc-component.ts
├── jsdoc-docurl.ts
├── jsdoc-helpers.ts
├── jsdoc-namespace.ts
└── jsdoc-useportal.ts
```

#### After

```
steps/serialization/component/implementation/typescript/comments
├── jsdoc-helpers.ts
└── jsdoc-uxpin-annotations.ts
```

## How to test

Run tests related to serialization to ensure nothing is broken:

```
npx jest src/steps/serialization/component/implementation
```
